superset/tasks/async_queries.py

```diff
@@ -43,6 +43,8 @@ def ensure_user_is_set(user_id: Optional[int]) -> None:
     user_is_set = hasattr(g, "user") and g.user is not None
     if not user_is_set and user_id is not None:
         g.user = security_manager.get_user_by_id(user_id)
+    else:
+        g.user = security_manager.get_anonymous_user()
```

Comment:
Wouldn't this set `g.user` to anonymous is `user_is_set`? It appears we should make this `elif not user_is_set:`?

Reply:
You are right, good catch!